From: CodeFetch <me@bibbl.com>
Date: Tue, 11 Oct 2016 09:06:22 +0200
Subject: [PATCH] Reintroduce HT40

Adds the site.conf option to set the HT mode.
---
 package/gluon-core/check_site.lua                        | 1 +
 package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/package/gluon-core/check_site.lua b/package/gluon-core/check_site.lua
index 480b367..1cb2b5d 100644
--- a/package/gluon-core/check_site.lua
+++ b/package/gluon-core/check_site.lua
@@ -27,6 +27,7 @@ for _, config in ipairs({'wifi24', 'wifi5'}) do
     need_string('regdom') -- regdom is only required when wifi24 or wifi5 is configured
 
     need_number(config .. '.channel')
+    need_string(config .. '.htmode')
 
     local rates = {1000, 2000, 5500, 6000, 9000, 11000, 12000, 18000, 24000, 36000, 48000, 54000}
     local supported_rates = need_array_of(config .. '.supported_rates', rates, false)
diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless b/package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless
index 6e80957..2c88579 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless
@@ -157,7 +157,7 @@ local function configure_radio(radio, index, config)
   uci:delete('wireless', radio, 'disabled')
 
   uci:set('wireless', radio, 'channel', channel)
-  uci:set('wireless', radio, 'htmode', 'HT20')
+  uci:set('wireless', radio, 'htmode', config.htmode)
   uci:set('wireless', radio, 'country', site.regdom)
 
   if config.supported_rates then
-- 
2.9.3
